### PR TITLE
[New Unit] Round UFO

### PR DESCRIPTION
**Pull request by Fireond**
Add round UFO unit.
Use mouse and w/s to move, better move experience. 
Shoot round distributed bullets.

### CODE REVIEW

File: src/battle_game/core/units/round_UFO.cpp [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+    auto diff = input_data.mouse_cursor_position - position_;
+    float new_rotation;
+    if (glm::length(diff) < 1e-4) {
+      new_rotation = rotation_;
+    } else {
+      new_rotation = std::atan2(diff.y, diff.x) - glm::radians(90.0f);
+    }
+    game_core_->PushEventRotateUnit(id_, new_rotation);
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/round_UFO.cpp:97]: This is a fake change, please inherit from unit::Tank if you are reusing the very same function.
Tell me the difference if there is one.

File: src/battle_game/core/units/round_UFO.cpp [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    }
+  }
+}
+
+bool RoundUFO::IsHit(glm::vec2 position) const {
+  position = WorldToLocal(position);
+  return position.x > -0.8f && position.x < 0.8f && position.y > -1.0f &&
+         position.y < 1.0f && position.x + position.y < 1.6f &&
+         position.y - position.x < 1.6f;
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/round_UFO.cpp:135]: As you have changed the model, this function should be changed as well.

File: src/battle_game/core/units/round_UFO.cpp [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+    game_core_->PushEventRotateUnit(id_, new_rotation);
+  }
+}
+
+void RoundUFO::TurretRotate() {
+  auto player = game_core_->GetPlayer(player_id_);
+  if (player) {
+    turret_rotation_ = rotation_;
+  }
+}
```
Comment (LazyJazz) [inline src/battle_game/core/units/round_UFO.cpp:104]: What is the purpose of this?

**Review by LazyJazz — CHANGES_REQUESTED** [review]

File: src/battle_game/core/units/round_UFO.cpp (hunk 1 above)
Comment (Fireond) [reply src/battle_game/core/units/round_UFO.cpp:97]: 变成了用ws前进后退鼠标控制旋转，没有使用ad

**Review by LazyJazz — DISMISSED** [review]
LGTM! Call me in wechat if you want to merge your contribution into the main branch.

**Review by LazyJazz — APPROVED** [review]
LGTM!

**Pull request merged**